src/mono/wasm/runtime/binding_support.js

```diff
@@ -256,9 +262,10 @@ var BindingSupportLib = {
 				return null;
 
 			// this could be JSObject proxy of a js native object
-			var js_handle = this._get_js_id (root.value);
-			if (js_handle > 0)
+			var js_handle = this._try_get_cs_owned_object_js_handle (root.value);
+			if (js_handle) {
```

Comment:
isn't the value of an invalid handle -1? in which case we need to do js_handle > 0

Reply:
If we receive -1 something is quite wrong here. It would mean that we got passed JSObject instance which was already disposed/collected.
I will add assert.